lwt/httpaf_lwt.mli

```diff
@@ -1,24 +1,48 @@
 open Httpaf
 
+module Buffer : sig
+  type t
+
+  val create : int -> t
+
+  val get : t -> f:(Bigstringaf.t -> off:int -> len:int -> int) -> int
+  val put : t -> f:(Bigstringaf.t -> off:int -> len:int -> int Lwt.t) -> int Lwt.t
+end
+
+module type IO = sig
+  type t
+
+  val read : t -> Buffer.t -> [> `Eof | `Ok of int ] Lwt.t
```

Comment:
I don't see a reason to tie implementations to providing a `read` call that uses a buffer. This should be implementable by providing a read that looks like this:

```ocaml
(** The region [[pos, pos + len)] is where read bytes can be written to *)
val read : t -> pos:int -> len:int -> [`Eof | `Ok of int ] Lwt.t
```

Code reuse of the function that turns this into what's actually used can go into the `Buffer` module.

Reply:
Wouldn't `read` in your example also need to take a bigarray that it read into?

Reply:
Yes.